pyccel/ast/literals.py

```diff
@@ -0,0 +1,134 @@
+""" This module contains all literal types
+"""
+from sympy               import Integer as sp_Integer
+from sympy               import Float as sp_Float
+from sympy.logic.boolalg import BooleanTrue as sp_BooleanTrue, BooleanFalse as sp_BooleanFalse
+from sympy.core.expr     import Expr
+
+from .basic              import PyccelAstNode, Basic
+from .datatypes          import (NativeInteger, NativeBool, NativeReal,
+                                  NativeComplex, NativeString, default_precision)
+
+__all__ = (
+    'LiteralBooleanTrue',
+    'LiteralBooleanFalse',
```

Comment:
I wonder if `LiteralTrue` and `LiteralFalse` should be singletons.
Maybe in the future we'll need one for each value of `precision` used in the code.

Reply:
That makes sense, but we might have a precision problem. See issue #538
Given that we already have this precision problem I could make them singletons for now. What do you think?

Reply:
Sorry for the late reply. Yes, please make them singletons!

Reply:
After some discussion we have realized that using singletons for `LiteralTrue` and `LiteralFalse` will complicate greatly our code if we ever need to change their precision in the semantic stage. Hence we have decided to avoid this change, and merge this PR as it is.